src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java

```diff
@@ -579,11 +579,15 @@ private String getOtherTab(String title) {
 
   private String getTabWithCondition(
       java.util.function.Predicate<? super Pair<String, String>> condition) {
-    Optional<String> newTab = driver.getWindowHandles().stream()
-        .map(handleName -> Pair.of(handleName, driver.switchTo().window(handleName).getTitle()))
-        .filter(condition).map(Pair::getKey).findFirst();
+    Optional<String> newTab = driver.getWindowHandles()
+      .stream()
+      .map(handleName -> Pair.of(handleName, driver.switchTo().window(handleName).getTitle()))
+      .peek(handleTitle -> PageObjectLogging.log("Found window", String.format("Window with title %s", handleTitle), true))
+      .filter(condition)
+      .map(Pair::getKey)
+      .findFirst();
     return newTab.orElseThrow(
-        () -> new NotFoundException(String.format("Tab with title %s doesn't exist", title)));
+        () -> new NotFoundException("Tab that satisfies the condition doesn't exist"));

```

Comment:
LGTM but, you can return immediately without the temporary `Optional<>` variable.